client/client.go

```diff
@@ -651,12 +651,25 @@ var MaxRetriesExceededError = errors.New("Tried maximum amount of times")
 // Simple algorithm: Start with one minute, and try three times, then double
 // interval (maxInterval is maximum) and try again. Repeat until we tried
 // three times with maxInterval.
-func GetExponentialBackoffTime(tried int, maxInterval time.Duration) (time.Duration, error) {
+func GetExponentialBackoffTime(tried int,
+	maxInterval time.Duration,
+	maxAttempts int) (time.Duration, error) {
 	const perIntervalAttempts = 3
 
 	interval := 1 * ExponentialBackoffSmallestUnit
 	nextInterval := interval
 
+	if maxAttempts > 0 {
+		if tried > maxAttempts {
+			return 0, MaxRetriesExceededError
+		}
+
+		for c := 0; c < tried; c++ {
+			interval *= 2
+		}
+		return interval, nil
+	}
```

Comment:
that was on purpose, thats what I understood from your comment: that we just try given number of times, not honouring the original scenario in case of maxAttempts given. since it is not maxAttempts, because we try 3 times each. ok. I will rework.

Reply:
It is using maxAttempts. I don't mean three *additional* times. With my proposed change, it follows maxAttempts exactly, *but* it does it by using the exponential fallback which is increased after each third attempt, like we have always done. It's possible it will never reach maxInterval in that case, but that was possible with your code as well.

Reply:
no problema, I immediately started fixing. I already pushed, tested, and updated the tests. I dropped `maxAttempts <= 0` I think we should return `MaxRetriesExceededError` at the last interval, even if the `maxAttempts` are set. agree?

Reply:
I think we should keep it because customers have complained about exactly this. That's also why I've pushed for having this change applied to existing retry operations as well. By adding the `RetryPollCount` flag, and allowing the number of retries to be set higher than what the default gives you, we give customers with long blackout windows, but short retry times, a way to avoid failing updates.

Reply:
ok re-instanted.